libraries/Microsoft.Bot.Builder/StreamingRequestHandler.cs

```diff
@@ -0,0 +1,207 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Connector;
+using Microsoft.Bot.StreamingExtensions;
+using Microsoft.Bot.StreamingExtensions.Transport;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Bot.Builder
+{
+    /// <summary>
+    /// Used to process incoming requests sent over an <see cref="IStreamingTransport"/> and adhering to the Bot Framework Protocol v3 with Streaming Extensions.
+    /// </summary>
+    public class StreamingRequestHandler : RequestHandler
+    {
+#if DEBUG
+        public
+#else
+        private
+#endif
+        string userAgent;
+
+        private IBot bot;
+
+        private IServiceProvider services;
+
+        private IList<IMiddleware> middlewareSet;
+
+        private Func<ITurnContext, Exception, Task> onTurnError;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="StreamingRequestHandler"/> class.
+        /// The StreamingRequestHandler serves as a translation layer between the transport layer and bot adapter.
+        /// It receives ReceiveRequests from the transport and provides them to the bot adapter in a form
+        /// it is able to build activities out of, which are then handed to the bot itself to processed.
+        /// Throws <see cref="ArgumentNullException"/> if arguments are null.
+        /// </summary>
+        /// <param name="onTurnError">Optional function to perform on turn errors.</param>
+        /// <param name="bot">The <see cref="IBot"/> to be used for all requests to this handler.</param>
+        /// <param name="middlewareSet">An optional set of middleware to register with the bot.</param>
+        public StreamingRequestHandler(Func<ITurnContext, Exception, Task> onTurnError, IBot bot, IList<IMiddleware> middlewareSet = null)
+        {
+            this.bot = bot ?? throw new ArgumentNullException(nameof(bot));
+            this.middlewareSet = middlewareSet;
```

Comment:
Check + assign variables in the same order they are passed in - unless there's a reason no to.